docs/blueprints/EP018.rst

```diff
@@ -0,0 +1,177 @@
+        URL
+
+.. code:: http
+
+    GET /api/kytos/auth/v1/login/ - This endpoint verifies a user and returns a valid token if authentication is correct.
+
+    $ curl -X GET \
+        -H 'Accept:application/json' \
+        -H 'Authorization:Basic username:password' \
+        URL
+
+.. code:: http
+
+    GET /api/kytos/auth/v1/users/ - This endpoint lists the registered users.
+
+    $ curl -X GET \
+        -H 'Accept:application/json' \
+        -H 'Authorization: Bearer ${TOKEN}' \
+        URL
+
+.. code:: http
+
+    GET /api/kytos/auth/v1/users/<user_id>/ - This endpoint gets details about a specific user.
+
+    $ curl -X GET \
+        -H 'Content-type:application/json' \
+        -H 'Accept:application/json' \
+        -H 'Authorization: Bearer ${TOKEN}' \
+        -d '{"user_id":"001"}' \
+        URL
+
+.. code:: http
+
+    DELETE /api/kytos/auth/v1/users/<user_id>/ - This endpoint delete a specific user.	
+
+    $ curl -X DELETE \
+        -H 'Content-type:application/json' \
+        -H 'Accept:application/json' \
+        -H 'Authorization: Bearer ${TOKEN}' \
+        -d '{"user_id":"001"}' \
+        URL
+
+.. code:: http
+
+    PATCH /api/kytos/auth/v1/users/<user_id>/ - This endpoint update a specific user.
+
+    $ curl -X PATCH \
+        -H 'Content-Type: application/json' \
+        -H 'Authorization: Bearer ${TOKEN}' \
+        -d '{"user_id":"001"}' \
+        URL
+
+Backwards Compatibility
+=======================
+
+This proposal doesn’t modify the existing project semantic, except for the
+inclusion of a new authentication module in kytos/core, with a decorator whose
+purpose is to control the restriction of access to rest endpoints. This new
+functionality makes the following dependencies mandatory: pyJWT and Storehouse
+NApp.
+
+Security Implications
+=====================
+
+The malicious user can capture a valid victim token and make requests on their
+behalf. One solution is that all endpoints must have HTTPS.
+
+How to Teach This
+=================
+
+Documentation should be changed to include these new features:
+
+    - For endpoints constraints, instructions should be added teaching users how to use @authenticated decorator.
+
+    - To list, create, and delete users, instructions should be added teaching users how to use the public endpoints of the authentication module:
+
+        - POST /api/kytos/auth/v1/users/ - Create a user.
+        - GET /api/kytos/auth/v1/login/ -  Authenticate a user.
+        - GET /api/kytos/auth/v1/users/ - List all users.
+        - GET /api/kytos/auth/v1/users/<user_id>/ - List specific user.
+        - DELETE /api/kytos/auth/v1/users/<user_id>/ - Delete specific user.
+        - PATCH /api/kytos/auth/v1/users/<user_id>/ - Update specific user.
+
+Open Issues
+===========
+
+- `Start Blueprint: Kytos API Authentication <https://github.com/kytos/kytos/issues/861>`_
+- `Create Kytos API Authentication blueprint <https://github.com/kytos/kytos/pull/955>`_
```

Comment:
Move these links to the References, and add here any questions that are still being discussed, if any.